bitbot/tarot.go

```diff
@@ -0,0 +1,119 @@
+package bitbot
+
+import (
+    "github.com/whyrusleeping/hellabot"
+    "math/rand"
+    "strings"
+    "strconv"
+)
+
+var TarotTrigger = NamedTrigger{
+    ID:   "Tarot",
+    Help: "Request tarot cards, default 1. Usage !tarot [num cards].",
+    Condition: func(irc *hbot.Bot, m *hbot.Message) bool {
+        return m.Command == "PRIVMSG" && strings.HasPrefix(m.Trailing, "!tarot")
+    },
+    Action: func(irc *hbot.Bot, m *hbot.Message) bool {
+        if (len(m.Content) < 7) {
+            resp := tarotCards[rand.Intn(len(magic8responses)-1)]
+            irc.Reply(m, resp)
+        } else {
+            msg := strings.TrimPrefix(m.Content, "!tarot ")
+            if num, err := strconv.Atoi(msg); err == nil {
+                if (num < 1 || num > len(tarotCards)) {
+                    num = 1
+                }
+                deck := rand.Perm(77)
+                card := 0
+                for i := 0; i < num; i++ {
+                    card, deck = deck[0], deck[1:]
+                    resp := tarotCards[card]
+                    irc.Reply(m, resp)
```

Comment:
There is actually no need to use `card` or pop and shift here. We can simply iterate our way across the generated permutation of integers.

```suggestion
                for i := 0; i < num; i++ {
                    irc.Reply(m, deck[i])
```

Reply:
i used `tarotCards[deck[i]]` instead.